umh-core/pkg/communicator/api/v2/http/requester.go

```diff
@@ -208,19 +209,11 @@ func enhanceConnectionError(err error) error {
 	return fmt.Errorf("connection error: %w (no response received from server, status code 0)", err)
 }
 
-// GetRequest does a GET request to the given endpoint, with optional header and cookies
-func GetRequest[R any](ctx context.Context, endpoint Endpoint, header map[string]string, cookies *map[string]string, insecureTLS bool, apiURL string, logger *zap.SugaredLogger) (result *R, statusCode int, responseErr error) {
-	// Set up context with default 30 second timeout if none provided
-	if ctx == nil {
-		var cancel context.CancelFunc
-		ctx, cancel = context.WithTimeout(context.Background(), 30*time.Second)
-		defer cancel()
-	}
-
-	url := apiURL + string(endpoint)
+// DoHTTPRequest performs the actual HTTP request and returns the response and any errors
+func DoHTTPRequest(ctx context.Context, url string, header map[string]string, cookies *map[string]string, insecureTLS bool, logger *zap.SugaredLogger) (*http.Response, error) {
 	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
 	if err != nil {
-		return nil, 0, err
+		return nil, err
```

Comment:
i remember the 0 was needed for some type of error handling

Reply:
The logic is still the same, see line 335